src/Symfony/Component/HttpClient/HttpOptions.php

```diff
@@ -0,0 +1,299 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * A helper providing autocompletion for available options.
+ *
+ * @see HttpClientInterface for a description of each options.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @experimental in 4.3
+ */
+class HttpOptions
+{
+    private $options = [];
```

Comment:
Is it intended to create extensions of this class, for adding custom options? If yes, `$options` should be protected, if no, the class could be final.

Reply:
could be - but no need to make it protected: toArray() provides everyone extensions could need.